components/elm/src/biogeophys/SedYieldMod.F90

```diff
@@ -128,71 +158,130 @@ subroutine SoilErosion (bounds, num_soilc, filter_soilc, &
             ! check landunit type and ground covered by snow/ice
             if ( lun_pp%itype(l)/=istsoil .and. lun_pp%itype(l)/=istcrop ) then
                cycle
-            else if (hsnow(c)>0) then
-               cycle
             end if
 
-            ! soil detachment by rainfall
+            ! check the glacier landunit in the gridcell
+            lg = grc_pp%landunit_indices(istice, g) 
+            fglacier = 1._r8
+            if (lg /= ispval) then
+               found = .false.
+               cg = lun_pp%coli(lg)
+               do while (.not. found .and. cg <= lun_pp%colf(lg))
+                  if (col_pp%active(cg)) then
+                     found = .true.
+                  else
+                     cg = cg + 1
+                  end if
+               end do
+               if (found .and. qflx_qrgwl(cg)<1e10_r8) then
+                  fglacier = 1._r8 + 9._r8 * lun_pp%wtgcell(lg)
+               end if
+            end if
+
+            ! tillage, lithology factors
+            ftillage = 2.7_r8 - 1.7_r8 * tillage(c)
+            flitho = litho(c)
+
+            ! ground cover calculated by plant residue
+            Brsd = 0.0_r8  ! kg/m2
+            do lp = 1, ndecomp_pools
+               if ( decomp_cascade_con%is_litter(lp) .and. decomp_cpools_vr(c,1,lp)>0._r8 ) then
+                  Brsd = Brsd + 1e-3_r8 * decomp_cpools_vr(c,1,lp) * dzsoi_decomp(1) 
```

Comment:
add decimal

Reply:
@tanzeli1982 please change line 178 from fglacier = 1._r8 + 9._r8 * lun_pp%wtgcell(lg) 
to 
fglacier = 1._r8 + 9._r8 * lun_pp%wttopounit(lt)